v21.2/crdb-internal.md

```diff
@@ -5,128 +5,642 @@ toc: true
+~~~sql
+SELECT  ti.descriptor_name as table_name, ti.index_name, total_reads
+FROM  crdb_internal.index_usage_statistics AS us
+JOIN crdb_internal.table_indexes ti
+ON us.index_id = ti.index_id AND us.table_id = ti.descriptor_id
+WHERE last_read < NOW() - INTERVAL '1 WEEK'
+ORDER BY total_access desc;
 ~~~
 
+##### Which indexes are infrequently used? Are there any unused indexes?
+
+{% include_cached copy-clipboard.html %}
+~~~sql
+SELECT  ti.descriptor_name as table_name, ti.index_name, total_reads
+FROM crdb_internal.index_usage_statistics AS us
+JOIN crdb_internal.table_indexes ti
+ON us.index_id = ti.index_id AND us.table_id = ti.descriptor_id
+WHERE total_reads = 0;
+~~~
+
+### `statement_statistics`
+
+Column | Type | Description
+------------|-----|------------
+`aggregated_ts` | `TIMESTAMPTZ NOT NULL` | The time that statistics aggregation started.
+`fingerprint_id` | `BYTES NOT NULL` | Unique identifier of the statement statistics. This is constructed using the statement fingerprint text, and statement metadata (e.g. query type, database name, etc.)
+`transaction_fingerprint_id` | `BYTES NOT NULL` | Uniquely identifies a transaction statistics. The transaction fingerprint ID that this statement statistic belongs to.
+`plan_hash` | `BYTES NOT NULL` | Uniquely identifies a query plan that was executed by the current statement. The query plan can be retrieved from the `sampled_plan` column.
+`app_name` | `STRING NOT NULL`| The name of the application that executed the statement.
+`metadata` | `JSONB NOT NULL` | Metadata that describes the statement. See [`metadata` column](#metadata-column).
+`statistics` | `JSONB NOT NULL` | Statistics for the statement. See [`statistics` column](#statistics-column).
+`sampled_plan` | `JSONB NOT NULL` | The sampled query plan of the current statement statistics. This column is unfilled if there is no sampled query plan.
+`aggregation_interval` | `INTERVAL NOT NULL` | The interval over which statistics are aggregated.
+
+### `metadata` column
+
+Column | Type | Description
+------------|-----|------------
+`db` | `STRING` | The database on which the statement is executed.
+`distsql` | `BOOLEAN` | Whether the statement is being executed by the Distributed SQL (DistSQL) engine.
+`failed` | `BOOLEAN` | Whether the statement execution failed.
+`fullScan` | `BOOLEAN` | Whether the statement performed a full scan of the table.
+`implicitTxn` | `BOOLEAN` | Whether the statement executed in an implicit transaction.
+`query` | `STRING` | The statement string.
+`querySummary` | `STRING` | The statement string summary.
+`stmtTyp` | `SQLType` | The type of statement: `TypeDDL`, `TypeDML`, `TypeDCL`, or `TypeTCL`.
+`vec` | `BOOLEAN` | Whether the statement executed in the vectorized query engine.
+
+### `statistics` column
+
+`execution_statistics` are the execution statistics for this statement.
+`statistics` are the statement statistics sent to the DB Console for a statement fingerprint.
```

Comment:
Hmm. Actually, both `statistics` field and `execution_statistics` field are sent to DB Console.

Reply:
Fixed.

Reply:
@Azhng, what's the differences between statistics and execution_statistics? They seem to be categorizing the same type of stats. Contention, Latency, etc.

I think I understand the following, "sent to the DB Console for a statement fingerprint means". But perhaps we should be more generic. The entire Statements page is reading information from the crdb_internal table. So something along the lines of, "The Statements overview and details pages (link to our console docs) pulls information from crdb_internal.statement_statistics". This can go right under the following [header](https://deploy-preview-12499--cockroachdb-docs.netlify.app/docs/v21.2/crdb-internal#statement_statistics). In case partners or customers want to replicate similar views in the console.

Reply:
`statistics`'s fields are always populated and they are updated each time a new statement of that statement fingerprint is executed. 

`execution_statistics` is collected via sampling. That is, we probablistically run a query with tracing enabled to collect more fine-grained statistics of the query execution. (Hence the name, `execution_statistics`)